### DISCUSSION TITLE
Recommended way to redirect

### DISCUSSION THREAD

**Discussion by alexandremjacques** [unpoly/unpoly#396] [Q&A]
Hi,

In the case of a form that is displayed in a new layer and, after submitted, redirects the user to the details page targeting the `up-main` (outside the layer)?

Is it possible to use `up.render()` with the redirected content from the server? Is it the recommended way?

TIA

**Comment by adam12** [comment]
I'm not sure I follow. How are you submitting the form?  Are you using an `[up-]` attribute? or doing it through the `up.` JS interface?

**Reply by alexandremjacques** [reply]
It a normal form with `up-submit` and a submit button. 
My question here is about how to redirect after form response since this form is inside a new layer and I want the response page do be rendered in a target (`up-main`) in the root layer (closing the form’s layer).

**Comment by adam12** [comment]
You could probably do it at least two ways:

1. Set an `[up-accept-location]` attribute on the `<a>` element that opens the modal, to signify which page marks the completion of the sub-interaction.
2. Set an `[up-layer="root"]` attribute on the `<form>` element performing the submission.

**Reply by alexandremjacques** [reply]
The first one will just accept (close) the form layer, right? At this moment, could `up-on-accept` call the `up.render()` passing the response content and target my `up-main` marked element? If so, how could I grab the response content to pass to `up-render()`?
In 2., `up-layer="root"` will target my `up-main` marked element or I shoul be specific about that? `up-layer="main"` would do?

TIA
